### PR TITLE
Merge 8.3.0

### PR DESCRIPTION
**Pull request by kamil-holubicki**

### CODE REVIEW

**Comment by venkatesh-prasad-v** [comment]
The below files have merge conflicts
```
$ grep "<<< HEAD" ./* -inR --exclude-dir="./build"
./mysql-test/suite/funcs_1/r/processlist_priv_ps.result:84:<<<<<<< HEAD
./storage/ndb/src/kernel/blocks/backup/Backup.cpp:7678:<<<<<<< HEAD
```

```
$ grep ">>> Perc" ./* -inR --exclude-dir="./build"
./azure-pipelines.yml:179:>>>>>>> Percona-Server-8.0.33-25
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Copyright (c) 2006, 2023, Oracle and/or its affiliates.
-# 
+#
```
Comment (venkatesh-prasad-v) [inline CMakeLists.txt]: There are many changes wrt spacing in this file. Is this a side-effect of IDE?

**Comment by venkatesh-prasad-v** [comment]
Running `./check_src/run.sh <path/to/PS-8.3-code>` reports many errors out of which there are some repeated errors due to renaming of `TABLE_LIST` to `Table_ref`

This makes the build fail for `-DWITH_WSREP=OFF`
```
In file included from /home/venki/work/pxc/merge/trunk/sql/auth/sql_auth_cache.h:53,
                 from /home/venki/work/pxc/merge/trunk/sql/auth/mysql_native_password.cc:30:
/home/venki/work/pxc/merge/trunk/sql/auth/auth_internal.h:218:33: error: ‘TABLE_LIST’ has not been declared
  218 | int open_grant_tables(THD *thd, TABLE_LIST *tables, bool *transactional_tables);
      |                                 ^~~~~~~~~~
```

Looks like `#define PFS_MAX_MUTEX_CLASS 350` is not defined in the `#else` part
```
============
Error: ./storage/perfschema/pfs_server.h
============
57a58
> #define PFS_MAX_MUTEX_CLASS 350
```

File: sql/sql_parse.cc [hunk 2]
```diff
@@ -573,13 +573,10 @@ void init_sql_command_flags() {
   server_command_flags[COM_STMT_RESET] |= CF_SKIP_WSREP_CHECK;
   server_command_flags[COM_STMT_SEND_LONG_DATA] |= CF_SKIP_WSREP_CHECK;
   server_command_flags[COM_QUIT] |= CF_SKIP_WSREP_CHECK;
-  server_command_flags[COM_PROCESS_INFO] |= CF_SKIP_WSREP_CHECK;
-  server_command_flags[COM_PROCESS_KILL] |= CF_SKIP_WSREP_CHECK;
```
Comment (venkatesh-prasad-v) [inline sql/sql_parse.cc]: Why this change? Is it because upstream removed `COM_PROCESS_INFO`, `COM_PROCESS_KILL` and `COM_PROCESS_KILL`

File: mysql-test/suite/galera/r/MW-362.result [hunk 3]
```diff
@@ -7,7 +7,7 @@ RESET BINARY LOGS AND GTIDS;
 SET SESSION WSREP_ON=ON;
 SHOW BINARY LOGS;
 Log_name	File_size	Encrypted
-mysqld-bin.000001	157	No
```
Comment (venkatesh-prasad-v) [inline mysql-test/suite/galera/r/MW-362.result]: Please check what caused this shift in binary log position by 1 byte.

File: mysql-test/suite/galera/t/galera_account_management3.test [hunk 4]
```diff
@@ -33,13 +33,13 @@ SET ROLE 'writer';
 GRANT ALL PRIVILEGES ON *.* TO 'user1'@'%' WITH GRANT OPTION;
 
 --let $binlog_file = mysqld-bin.000002
---let $binlog_position = 2282
```
Comment (venkatesh-prasad-v) [inline mysql-test/suite/galera/t/galera_account_management3.test]: Likewise

File: storage/perfschema/pfs_server.h [hunk 5]
```diff
@@ -55,7 +55,18 @@
...
+   Max value for PXC builds.
+   WSREP patch defines 39 additional PSI_mutex_key keys:
+   key_COND_galera_* => 24
+   key_COND_wsrep_* => 15
+   Let's keep PFS_MAX_MUTEX_CLASS always bigger by this number.
+*/
+#define PFS_WSREP_MUTEX_CLASS 39
+#define PFS_MAX_MUTEX_CLASS (350 + PFS_WSREP_MUTEX_CLASS)
+#else
+#endif /* WITH WSREP */ /* PFS_MAX_MUTEX_CLASS */
```
Comment (venkatesh-prasad-v) [inline storage/perfschema/pfs_server.h]: PFS_MAX_MUTEX_CLASS will not be defined when WITH_WSREP=OFF
```suggestion c++
#else
#define PFS_MAX_MUTEX_CLASS 350
#endif /* WITH WSREP */ /* PFS_MAX_MUTEX_CLASS */
```

**Review by venkatesh-prasad-v — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (kamil-holubicki) [reply CMakeLists.txt]: fixed

File: sql/sql_parse.cc (hunk 2 above)
Comment (kamil-holubicki) [reply sql/sql_parse.cc]: yes

File: storage/perfschema/pfs_server.h (hunk 5 above)
Comment (kamil-holubicki) [reply storage/perfschema/pfs_server.h]: fixed

**Comment by kamil-holubicki** [comment]
Conflicts:
> ./mysql-test/suite/funcs_1/r/processlist_priv_ps.result:84:<<<<<<< HEAD
Fixed. We do not run tests with --ps-protocol, so it was skipped and not detected. I think we should consider running tests with --ps-protocol as well.

> ./storage/ndb/src/kernel/blocks/backup/Backup.cpp:7678:<<<<<<< HEAD
Old PS issue. Let's leave it as it is.

> ./azure-pipelines.yml:179:>>>>>>> Percona-Server-8.0.33-25
Fixed. Old conflict from resolve artifact from 8.0.33 merge time.

File: mysql-test/suite/galera/r/MW-362.result (hunk 3 above)
Comment (kamil-holubicki) [reply mysql-test/suite/galera/r/MW-362.result]: It is related to the introduction of the new event `GTID_TAGGED_LOG_EVENT`. Now `FORMAT_DESCRIPTION_EVENT` header is 1 byte longer.

(c1fcd5f1b73709094e86474e2406ed89b83109e2)

**Review by venkatesh-prasad-v — APPROVED** [review]
LGTM
